src/stack.ts

```diff
@@ -0,0 +1,29 @@
+import { VariableStatementAnalysis } from "./VariableStatementAnalysis";
+interface IStack<T> {
+    push(item: T): void;
+    pop(): T | undefined;
+    peek(): T | undefined;
+    size(): number;
+}
+
+export class Stack<T> implements IStack<T> {
+    private storage: T[] = [];
+
+    constructor(mainDetectedVariableStatements: VariableStatementAnalysis[]) {}
```

Comment:
I like the generic stack! :) 

What is this parameter in the constructor for?

Reply:
Since it will be used for scoping, variables defined in the general scope can be used in a specific scope (like an if statement)
So i want to pass the variables of the previous scope to the new scope and maybe create a new array that collects the variables of the new scope or something like that
unless you have a different idea on how to use the stack!